src/Sylius/Bundle/CoreBundle/EventListener/Workflow/Order/CancelOrderPaymentListener.php

```diff
@@ -31,6 +31,8 @@ public function __invoke(CompletedEvent $event): void
         $order = $event->getSubject();
         Assert::isInstanceOf($order, OrderInterface::class);
 
-        $this->compositeOrderStateMachine->apply($order, OrderPaymentTransitions::GRAPH, OrderPaymentTransitions::TRANSITION_CANCEL);
+        if ($this->compositeOrderStateMachine->can($order, OrderPaymentTransitions::GRAPH, OrderPaymentTransitions::TRANSITION_CANCEL)) {
```

Comment:
Could you please explain why, from now in case of the exception we are choosing to do nothing instead of raising the exception?

Reply:
I've replicated the functionality as it was previously implemented in winzu state machine. Details are in the description of this PR. :stuck_out_tongue: